site/content/languages/go.md

```diff
@@ -241,6 +241,52 @@ signature unrelated to the source, more care is needed implementing the host
 side, to ensure the proper count of parameters are read and results written to
 the Go stack.
 
+## Hacking
+
+If you run into an issue where you need to change Go's sourcecode, the first
+thing you should do is read the [contributing guide][20], which details how to
+confirm an issue exists and a fix would be accepted. Assuming they say yes, the
+next step is to ensure you can build and test go.
+
+### Setting up your GOROOT
+
+First, clone upstream or your fork of golang/go and make a branch off `master`
+for your work, as GitHub pull requests are against that branch.
+
+```bash
+$ git clone https://github.com/golang/go.git
+$ cd go
+$ export GOROOT=$PWD
+$ git checkout -b my-fix
+```
+
+### Build a branch-specific `go` binary
+
+While your change may not affect the go binary itself, there are checks inside
+go that require version matching. Build a go binary from source to avoid these:
+```bash
+$ cd src
+$ GOOS=js GOARCH=wasm ./make.bash
```

Comment:
Do these variables work? They look like they would create a compiler that runs in the browser, not on the computer

Reply:
PTAL redid it

Reply:
Cool - also found `GOHOSTARCH` so makes sense now https://github.com/golang/go/blob/master/src/make.bash#L13